<h2>Measures of Central Tendency</h2>

What are Measures of Central Tendency?

Measures of Central Tendency are statistical measures that represent the center or typical value of a dataset. They summarize a large amount of data into a single representative value.

<h2>Why are they important?</h2>

<ul><li>Summarize large datasets.</li>
<li>Compare different datasets.</li>
<li>Identify the typical value.</li>
<li>Used in Business Analytics, Finance, Healthcare, and AI/ML.</li></ul>
    
<h2>The most common measures are:</h2>

<ul><li>Mean</li>
<li>Median</li>
<li>Mode</li>
<li>Weighted Mean</li>
<li>Geometric Mean</li>
<li>Harmonic Mean</li></ul>

<h2>Mean</h2>

Mean is the average of all values in a dataset.

In simple words:

Add all values and divide by the number of values.

Mean = n∑x

x = each value

n = number of values

Example:

Marks of five students: 70, 80, 90, 85, 75

Sum = 70 + 80 + 90 + 85 + 75 = 400

Mean = 400 / 5 = 80

Mean = 80

In [1]:
import numpy as np

marks = [70, 80, 90, 85, 75]

print(np.mean(marks))

80.0


<h2>Median</h2>

The Median is the middle value of a dataset after arranging the values in ascending order.

If there are:

Odd number of values → middle value Even number of values → average of two middle values

Formula:

For odd observations

Middle value

For even observations

Median = middle 1 + middle 2 / 2

Odd Example:

Dataset: 12, 18, 20, 25, 30

Already sorted

Middle value = 20

Median = 20

Even example:

Dataset: 10, 20, 30, 40

Median: (20 + 30)/2 = 25

In [2]:
import numpy as np

numbers = [12, 18, 20, 25, 30]

print(np.median(numbers))

20.0


<h2>Mode</h2>

The Mode is the value that appears most frequently in a dataset.

A dataset may have:

One mode (Unimodal) Two modes (Bimodal) Multiple modes (Multimodal) No mode

Formula:

No mathematical formula.

It is simply the value with the highest frequency.

Numerical Example

Dataset: 2, 4, 5, 5, 6, 7, 5

Frequency: 2, 4, 5, 5, 6, 7, 5

Mode = 5

In [3]:
import pandas as pd

numbers = pd.Series([2,4,5,5,6,7,5])

print(numbers.mode())

0    5
dtype: int64


<h2>Weighted Mean</h2>

<h3>What is Weighted Mean?</h3>

Weighted Mean is an average where some values are given more importance (weight) than others.

In a normal mean, every value has equal importance.
    
In a weighted mean, different values can have different importance.

Formula

Weighted Mean= ∑(Value×Weight) / ∑Weight 


Where:

x = value w = weight

Example:

| Subject    | Mark | Weight |
| ---------- | ---: | -----: |
| Python     |   80 |      3 |
| SQL        |   70 |      2 |
| Statistics |   90 |      1 |

Calculation :

(90×4)+(80×3)+(70×2)=740

Total credits=9

Weighted Mean=740/9=82.22

In [4]:
import numpy as np

marks = [90,80,70]
weights = [4,3,2]

print(np.average(marks, weights=weights))

82.22222222222223


Weighted Mean is an average in which each value contributes according to its assigned importance or weight.

<h2>Geometric Mean</h2>

What is Geometric Mean?

Geometric Mean is a type of average used when values are multiplied together rather than simply added.

Geometric Mean= nVx1×x2×x3×...×xn

Step 1: Multiply all the values

Step 2: Find the root based on the number of values

Example:

Numbers-2,4,8

Product = 2×4×8 = 64

Product = 2×4×8 = 64

Geometric Mean = 4

In [6]:
!pip install scipy

   ---------------------------------------- 0.0/37.4 MB ? eta -:--:--
   ---------------------------------------- 0.0/37.4 MB ? eta -:--:--
   ---------------------------------------- 0.0/37.4 MB ? eta -:--:--
   ---------------------------------------- 0.3/37.4 MB ? eta -:--:--
   ---------------------------------------- 0.3/37.4 MB ? eta -:--:--
    --------------------------------------- 0.5/37.4 MB 661.4 kB/s eta 0:00:56
    --------------------------------------- 0.8/37.4 MB 895.7 kB/s eta 0:00:41
   - -------------------------------------- 1.0/37.4 MB 1.0 MB/s eta 0:00:37
   - -------------------------------------- 1.3/37.4 MB 1.0 MB/s eta 0:00:36
   - -------------------------------------- 1.6/37.4 MB 1.0 MB/s eta 0:00:36
   - -------------------------------------- 1.8/37.4 MB 994.9 kB/s eta 0:00:36
   - -------------------------------------- 1.8/37.4 MB 994.9 kB/s eta 0:00:36
   -- ------------------------------------- 2.4/37.4 MB 1.1 MB/s eta 0:00:34
   -- --------------------

In [7]:
from scipy.stats import gmean

numbers = [2,4,8]

print(gmean(numbers))

4.0


Imagine a company grows its sales:

Year 1 - 2 times

Year 2 - 8 times

For multiplicative growth factors, geometric mean is useful because the changes multiply over time.

It is commonly used when analyzing growth rates, investment returns, and ratios.

<h2>Harmonic Mean</h2>

The Harmonic Mean is the reciprocal of the arithmetic mean of the reciprocals of the observations.

It is useful for rates such as speed, price per unit, or ratios.

Harmonic Mean - Useful when dealing with rates or "per" values.

Examples:

km/hour - speed

tasks/hour - productivity

requests/second - system performance

Precision and Recall - F1-score in AI/ML

Formula:

Harmonic Mean=n / (1 / x1) + (1 / x2) +...+ (1 / xn)

Example :

Speeds= 40 km/h, 60 km/h

Calculation:

HM = 2 / (1/40 + 1/60)

HM = 48 km/h

In [9]:
from scipy.stats import hmean

numbers = [40,60]

print(hmean(numbers))

47.99999999999999


Usage:

The F1-score, a widely used classification metric, is the harmonic mean of Precision and Recall.

For example, if:

Precision = 0.80 Recall = 0.60